### ISSUE TITLE
self-bound delegates CTD.

### ISSUE THREAD

**Issue by mlheur** [KSP-KOS/KOS#2918]
I'm building a set of libraries, trying to mimic some OOP concepts, and predictably crashing to desktop when accessing delegates in certain ways e.g. printing the object that contains the self-bound delegate.

```
@LAZYGLOBAL OFF.
// instantiate the object
local the_object is lex("state", "some state").
// define the member function
local function show_state {parameter self.print "self state: "+self:state.}.
// add the member function to the object instance.
the_object:add("show_state",show_state@:bind(the_object)).
// use the member function.
the_object:show_state().
wait until false. // comment this line to crash your program.
print the_object.  // CTD for accessing a self-bound member.
// maybe due to infinite recursion following the member looping back to the parent.
```

**Comment by thexa4** [comment]
This seems like it's the same as #1598. A fix has been submitted in #2882.

**Comment by mlheur** [comment]
Thanks, tried searching but didn't find those.
Also, with my above code, is not so predictable.
Closing for now.
